loader/loader_test.go

```diff
@@ -0,0 +1,423 @@
+			Expect(returnedObject.Name).To(Equal(releasePlanAdmission.Name))
+		})
+
+		It("fails to return an active release plan admission if the release plan does not match", func() {
+			modifiedRelease := release.DeepCopy()
+			modifiedRelease.Spec.ReleasePlan = "non-existent-release-plan"
+
+			returnedObject, err := loader.GetActiveReleasePlanAdmissionFromRelease(ctx, k8sClient, modifiedRelease)
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(BeTrue())
+			Expect(returnedObject).To(BeNil())
+		})
+	})
+
+	Context("When calling GetApplication", func() {
+		It("returns the requested application", func() {
+			returnedObject, err := loader.GetApplication(ctx, k8sClient, releasePlanAdmission)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&applicationapiv1alpha1.Application{}))
+			Expect(returnedObject.Name).To(Equal(application.Name))
+		})
+	})
+
+	Context("When calling GetApplicationComponents", func() {
+		It("returns the requested list of components", func() {
+			returnedObjects, err := loader.GetApplicationComponents(ctx, k8sClient, application)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObjects).To(HaveLen(1))
+			Expect(returnedObjects[0].Name).To(Equal(component.Name))
+		})
+	})
+
+	Context("When calling GetEnterpriseContractPolicy", func() {
+		It("returns the requested enterprise contract policy", func() {
+			returnedObject, err := loader.GetEnterpriseContractPolicy(ctx, k8sClient, releaseStrategy)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&ecapiv1alpha1.EnterpriseContractPolicy{}))
+			Expect(returnedObject.Name).To(Equal(enterpriseContractPolicy.Name))
+		})
+	})
+
+	Context("When calling GetEnvironment", func() {
+		It("returns the requested environment", func() {
+			returnedObject, err := loader.GetEnvironment(ctx, k8sClient, releasePlanAdmission)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&applicationapiv1alpha1.Environment{}))
+			Expect(returnedObject.Name).To(Equal(environment.Name))
+		})
+	})
+
+	Context("When calling GetRelease", func() {
+		It("returns the requested release", func() {
+			returnedObject, err := loader.GetRelease(ctx, k8sClient, release.Name, release.Namespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1alpha1.Release{}))
+			Expect(returnedObject.Name).To(Equal(release.Name))
+		})
+	})
+
+	Context("When calling GetReleasePipelineRun", func() {
+		It("returns a PipelineRun if the labels match with the release data", func() {
+			returnedObject, err := loader.GetReleasePipelineRun(ctx, k8sClient, release)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1beta1.PipelineRun{}))
+			Expect(returnedObject.Name).To(Equal(pipelineRun.Name))
+		})
+
+		It("fails to return a PipelineRun if the labels don't match with the release data", func() {
+			modifiedRelease := release.DeepCopy()
```

Comment:
this should be deleted like line 95 right?

Reply:
Nope. No object is created here.